clueless/state.py

```diff
@@ -41,33 +51,63 @@ def sock_addr(self) -> str:
 
     @property
     def is_registered(self) -> bool:
-        return self.character is not None
+        return self.state is UserState.REGISTERED
 
     async def dispatch_message(self, message):
+        print("Dispatch_message function:")
         if isinstance(message, CreateGame):
-            self.logger.debug(f"Received crate game")
-            game = GameState()
-            id = game.id
-            self.logger.debug(f'Created game {id}')
-            GAMES[id] = game
-            await GAMES[id].add_user(self)
+            if self.game is None:                       # If this player instance is not in a game
+                self.game = GameState()                 # Create an instance of GameState class, associate it with Player instance
+                id = self.game.id                            
+                self.logger.debug(f'Created game {id}')
+                GAMES[id] = self.game                   # Place in a dictionary of games
+                await GAMES[id].add_user(self)          # Add user into recently created class
+            else:
+                self.send_message(Status("Player is already in a game")) # RuntimeWarning: coroutine 'Player.send_message' was never awaited
```

Comment:
this needs an `await` before it. The `send_message` fn was declared as an asynchronous fn (`async def`) so "calling" it like this does not actually call it, it just sets up the state to call it creating a "future" you need to prefix this with `await` to execute that future and actually call the function.